src/main/java/by/triumgroup/recourse/service/impl/AbstractCrudService.java

```diff
@@ -44,4 +44,9 @@ protected AbstractCrudService(CrudRepository<E, ID> repository) {
         }
         return result;
     }
+
+    @Override
+    public Iterable<E> findAll() {
+        return wrapJPACall(() -> repository.findAll());
```

Comment:
couldn't. `repository.findAll()` is overloaded and can accept `WrapperVoidFunction` and `WrapperFunction<Iterable<E>>`, so need casting here. `wrapJPACall((WrapperFunctions.WrapperFunction<Iterable<E>>) repository::findAll)` looks ugly, isn't it?